setup.md

```diff
@@ -4,166 +4,241 @@ This document will guide you through setting up Fennel on your
+directory, you can run `fennel filename.fnl` to run a Fennel file.
+
+## Using LuaRocks to download and install Fennel
 
 [LuaRocks](https://luarocks.org/) contains a repository of Lua
-packages. It allows you to download and install packages and automates
-installation steps.
+software packages. LuaRocks is convenient because it automates the
+downloading, installation, and uninstallation of software packages.
+
+### To download and install Fennel
 
-1. Open up a shell
-2. Ensure the `luarocks` package is installed on your system
-3. Ensure the `~/.luarocks/bin` directory is added to your shell's `$PATH`
-4. Run `luarocks --local install fennel`
+1. Ensure the `luarocks` package is installed on your system
+2. Ensure the `~/.luarocks/bin` directory is added to your shell's `$PATH`
+3. Run `luarocks --local install fennel`
 
-You can try running `fennel --help` to confirm the installation succeeded.
+**Note**: You can try running `fennel --help` to confirm the
+installation succeeded.
 
 # Embedding Fennel
 
-Installing Fennel on your system will allow you to run scripts written
-in pure Fennel. However, for more complex situations it's common to
-embed Fennel code inside a larger application. You can do this in two
-ways: including the Fennel compiler or performing ahead-of-time compilation.
+Fennel can be embedded inside of Lua applications by including the
+Fennel compiler inside of a Lua application, or by performing
+ahead-of-time compilation.
 
-Including the Fennel compiler is a much more flexible way to go, and
-it's recommended that you go this route if you can. This allows you to
-offer extensibility features that let your users write their own
-Fennel scripts to automate your application. However, if you are
-working with an application that has more restrictions, it may be
-simpler to compile your Fennel code to Lua during the build process
-and only include the Lua output in the application.
+This section consists of the following subsections:
 
-## Embedding the compiler
+* [Embedding the Fennel compiler in a Lua application](#embedding-the-fennel-compiler-in-a-lua-application)
+* [Performing ahead-of-time compilation](#performing-ahead-of-time-compilation)
 
-Add `fennel.lua` to your repository, then you can load it from Lua like so:
+**Note**: Embedding the Fennel compiler in an application is the more
+flexible option, and is recommended. By embedding the Fennel compiler
+in an application, users can write their own extension scripts in
+Fennel to interact with the application. If the application is more
+restricted, then compiling Fennel code to Lua during the build
+process, and including the Lua output in the application may be
+easier.
+
+## Embedding the Fennel compiler in a Lua application
+
+The Fennel compiler can be added to your code repository, and then
+loaded from Lua. This allows you to call functions defined in Fennel
+code from Lua, and functions defined in Lua from Fennel without
+slowdown.
```

Comment:
I realized that the bit about being able to call functions defined in Fennel from Lua and the other way around maybe shouldn't go in this section, because it's not specific to embedding the Fennel compiler; it is true in both cases.

Reply:
Another option could be throwing that information into a "note", as it's additional information, but still related. Would you like to do this or remove it as you suggested? :D

Reply:
Ahhh I see what you meant now. How about I move it inside the main parent "# Embedding Fennel"?

Reply:
Chatted with @Benaiah about this, and have settled on removing it for now.